x-pack/plugins/timelines/public/components/t_grid/toolbar/fields_browser/field_table_header.test.tsx

```diff
@@ -0,0 +1,119 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { TestProviders } from '../../../../mock';
+import { FieldTableHeader, FieldTableHeaderProps } from './field_table_header';
+
+const mockSetFilterSelectedEnabled = jest.fn();
+const defaultProps: FieldTableHeaderProps = {
+  fieldCount: 0,
+  filterSelectedEnabled: false,
+  setFilterSelectedEnabled: mockSetFilterSelectedEnabled,
+};
+
+describe('FieldTableHeader', () => {
+  describe('FieldCount', () => {
+    it('should render empty field table', () => {
+      const result = render(
+        <TestProviders>
+          <FieldTableHeader {...defaultProps} />
+        </TestProviders>
+      );
+
+      expect(result.getByTestId('fields-showing').textContent).toBe('Showing 0 fields');
+    });
+
+    it('should render field table with one singular field count value', () => {
+      const result = render(
+        <TestProviders>
+          <FieldTableHeader {...defaultProps} fieldCount={1} />
+        </TestProviders>
+      );
+
+      expect(result.getByTestId('fields-showing').textContent).toBe('Showing 1 field');
+    });
+    it('should render field table with multiple fields count value', () => {
+      const result = render(
+        <TestProviders>
+          <FieldTableHeader {...defaultProps} fieldCount={4} />
+        </TestProviders>
+      );
+
+      expect(result.getByTestId('fields-showing').textContent).toBe('Showing 4 fields');
+    });
+  });
+
+  describe('View selected', () => {
+    beforeEach(() => {
+      mockSetFilterSelectedEnabled.mockClear();
+    });
+
+    it('should render "view all" option when filterSelected is not enabled', () => {
+      const result = render(
+        <TestProviders>
+          <FieldTableHeader {...defaultProps} filterSelectedEnabled={false} />
+        </TestProviders>
+      );
+
+      expect(result.getByTestId('viewSelectorButton').textContent).toBe('View: all');
+    });
+
+    it('should render "view selected" option when filterSelected is not enabled', () => {
+      const result = render(
+        <TestProviders>
+          <FieldTableHeader {...defaultProps} filterSelectedEnabled={true} />
+        </TestProviders>
+      );
+
+      expect(result.getByTestId('viewSelectorButton').textContent).toBe('View: selected');
+    });
+
+    it('should open the view selector with button click', async () => {
+      const result = render(
+        <TestProviders>
+          <FieldTableHeader {...defaultProps} />
+        </TestProviders>
+      );
+
+      expect(result.queryByTestId('viewSelectorMenu')).toBeNull();
+      expect(result.queryByTestId('viewSelectorOption-all')).toBeNull();
+      expect(result.queryByTestId('viewSelectorOption-selected')).toBeNull();
```

Comment:
nit:
```suggestion
      expect(result.queryByTestId('viewSelectorMenu')).not.toBeInTheDocument();
      expect(result.queryByTestId('viewSelectorOption-all')).not.toBeInTheDocument();
      expect(result.queryByTestId('viewSelectorOption-selected')).not.toBeInTheDocument();
```

Reply:
Oh, I tried something like this in other places and it crashed the test, so I never used `.not.toBeInTheDocument` again lol. Maybe I was using the wrong method... 🤷‍♂️  Good to know it works with `queryBy`. Thanks, I'll take it!